syslog-ng-rs/syslog-ng-common/src/formatter.rs

```diff
@@ -11,14 +11,12 @@ use std::fmt::Write;
 /// Applies transformations to key-value pairs.
 #[derive(Clone)]
 pub struct MessageFormatter {
-    buffer: String,
```

Comment:
A second best choice would be using thread local storage. I hope this new allocation will not introduce a bottleneck. Could you perhaps do a quick profiling to verify?